### PR TITLE
Added support for kong v0.13

### PR DESCRIPTION
**Pull request by sravanmedarapu**
- Added support for kong v0.13

### CODE REVIEW

**Comment by IvanoffDan** [comment]
With 0.13 Routes & Services have replaced APIs. Thus, adding support for 0.13 goes beyond adjusting schema. I'm currently working on a PR but do not have an ETA yet.

**Comment by aroundthecode** [comment]
anyone taking care of this ? looking forward for for support for 0.13 :)

**Comment by PGBI** [comment]
@aroundthecode kind of busy right now, so I haven't taken the time to review this yet. Will try to do it sometime this week. I see the tests are failing though. Did you check out the branch and test it? Does it work?

**Comment by sravanmedarapu** [comment]
@PGBI, I have tested locally its working fine. I could see kong v0.10 test cases are failing. But my repo Travis builds [passed](https://travis-ci.org/sravanmedarapu/kong-dashboard), just to make sure it's not a false failure would request you to rerun the build(Rerun build is not enabled for me).

**Comment by naturallight** [comment]
+1

**Comment by sravanmedarapu** [comment]
@PGBI, Travis build passing now.

**Comment by smileisak** [comment]
Is there any plans to merge this PR ?

**Comment by fangchih** [comment]
I found the current version cannot support the new objects (services, routes) in kong 0.13, will these new objects impact the design of dashboard?

**Comment by peeyush1234** [comment]
@IvanoffDan , Yes, the Kong 0.13 has routes and services now. However, 0.13 is still backward compatible with APIs ([link](https://github.com/Kong/kong/blob/master/CHANGELOG.md#core-2)). APIs will only go away in future relases of Kong. Hence, @sravanmedarapu PR is still valid, which will enable us to use Kong 0.13 with the kong-dashbaord. However, there will be work required to add the support for routes and services. (@IvanoffDan & @PGBI ) I can start the work for adding support for Routes and Service if none of you have started working on it.

My team is crruntely in process of migrating ot Kong 0.13 and would like to keep using kong-dashboard. @PGBI I would appreciate, if you can take a look at this PR and merge in to 3.X.X release.

@IvanoffDan, please let me know if you think otherwise.

File: bin/kong-dashboard.js [hunk 1]
```diff
@@ -162,8 +162,8 @@ function start(argv) {
       terminal.error("This version of Kong dashboard doesn't support Kong v0.9 and lower.");
       process.exit(1);
     }
-    if (semver.gte(version, '0.13.0')) {
-      terminal.error("This version of Kong dashboard doesn't support Kong v0.13 and higher.");
+    if (semver.gte(version, '0.14.0')) {
+      terminal.error("This version of Kong dashboard doesn't support Kong v0.14 and higher.");
```
Comment (peeyush1234) [inline bin/kong-dashboard.js:166]: The messaging can be improved.

Since this PR does not support Routes and Services, I would suggest adding some banner/info in the UI to tell the users that kong-dashboard does not support Routes and Services yet. How about adding it in the [footer](https://github.com/PGBI/kong-dashboard/blob/55f793b844c02274b2825db1b62668f376c8f10b/src/html/index.html#L54)?

File: .travis.yml [hunk 2]
```diff
@@ -16,6 +16,7 @@ env:
   - KONG_VERSION=0.10
   - KONG_VERSION=0.11
   - KONG_VERSION=0.12
+  - KONG_VERSION=0.13.0
```
Comment (peeyush1234) [inline .travis.yml]: Why not use "0.13" instead of 0.13.0?

File: package.json [hunk 3]
```diff
@@ -1,8 +1,8 @@
 {
-  "name": "kong-dashboard",
+  "name": "kong-dashboard-tmp",
   "description": "Web UI for managing Kong gateway",
   "keywords": "UI, kong, gateway",
-  "version": "3.1.0",
+  "version": "0.1.199",
```
Comment (peeyush1234) [inline package.json]: Why did you change the version to 0.1.199?

File: package.json [hunk 4]
```diff
@@ -1,8 +1,8 @@
 {
-  "name": "kong-dashboard",
+  "name": "kong-dashboard-tmp",
```
Comment (peeyush1234) [inline package.json]: It should be 'kong-dashboard' only.

File: tests/run.js [hunk 5]
```diff
@@ -18,5 +18,6 @@ request.get('http://localhost:8001').then((response) => {
   var kongVersion = semver.major(version) + '.' + semver.minor(version);
   return execSync('KONG_VERSION=' + kongVersion + ' node_modules/.bin/protractor tests/conf.js', {stdio: 'inherit'});
 }).catch((error) => {
+	terminal.error(error);
```
Comment (peeyush1234) [inline tests/run.js]: I would fix the indentation.

**Review by peeyush1234 — COMMENTED** [review]
@sravanmedarapu I have few comments on your code.

File: package-lock.json [hunk 6]
```diff
@@ -1,6 +1,6 @@
 {
-  "name": "kong-dashboard",
-  "version": "3.1.0",
+  "name": "kong-dashboard-tmp",
+  "version": "0.1.199",
```
Comment (PGBI) [inline package-lock.json]: What's going on here?
Please don't modify the version yourself, I'll take care of that after merging your PR.

File: .travis.yml (hunk 2 above)
Comment (PGBI) [reply .travis.yml]: I agree, 0.13 should be used. It makes sure tests are always run against the last patched version.

**Comment by PGBI** [comment]
@sravanmedarapu Thanks for your PR and sorry for this very long review delay...

Once you've taken care of the inline comments, I'll happily merge.

I'll update the Readme file to emphasize the fact that Kong-Dashboard 3.2.0 will be only partially compatible with Kong 0.13 since it doesn't support routes and services yet.

@peeyush1234 I haven't started implementing routes & services. I'll make sure this PR is merged soon so that you can tackle routes & services if you still want to.

**Comment by sravanmedarapu** [comment]
@PGBI, tests are failing by updating kong version to 0.13 from 0.13.0. I am running busy this week, would appreciate if anyone would like to check this.

**Comment by peeyush1234** [comment]
@PGBI I have been working on the routes and services. Here is my work in progress code (https://github.com/peeyush1234/kong-dashboard/pull/1). Hopefully, I will put out a pull request against the main repo early next week.

**Comment by PGBI** [comment]
@sravanmedarapu Basic auth plugin test was not compatible with Kong 0.13. I fixed it.
